internal/contractgateway/smartcontractgw.go

```diff
@@ -216,7 +216,12 @@ func (g *smartContractGW) PostDeploy(msg *messages.TransactionReceipt) error {
 				err = g.cs.AddRemoteInstance(msg.RegisterAs, "0x"+addrHexNo0x)
 			}
 		} else {
-			_, err = g.cs.AddContract(addrHexNo0x, requestID, registeredName, msg.RegisterAs)
+			abiID := requestID
+			if msg.Headers.ReqABIID != "" {
```

Comment:
So there's still a case where we expect this header to be missing, and we fall back to using the requestID?